### PR TITLE
Disable checkpoint conversion inside AutoResume

### PR DESCRIPTION
**Pull request by hemildesai**
# What does this PR do ?

Checkpoint conversion is not feasible inside AutoResume as it is not reliable in multi process/multi node scenarios. This update disables checkpoint conversion inside AutoResume and forces users to use a NeMo based checkpoint.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/lightning/megatron_parallel.py [hunk 1]
```diff
@@ -426,16 +426,20 @@ def init_ddp(self):
...
-                module,
-                data_parallel_group=parallel_state.get_data_parallel_group(with_context_parallel=True),
-                expert_data_parallel_group=parallel_state.get_data_modulo_expert_parallel_group(),
-                # Turn off bucketing for model_chunk 2 onwards, since communication for these
-                # model chunks is overlapped with compute anyway.
-                disable_bucketing=(model_chunk_idx > 0),
-            )
+            # Mcore DistributedDataParallel has to be called with grad. Normally this call is redundant, but for
+            # PEFT with num_sanity_val_steps > 0 this is necessary.
+            with torch.enable_grad():
```
Comment (hemildesai) [inline nemo/lightning/megatron_parallel.py:431]: Should there be a check to only use this if it's PEFT?

Comment (cuichenx) [reply nemo/lightning/megatron_parallel.py]: thanks for the comment, revised!

**Comment by akoumpa** [comment]
@hemildesai this is great, have you run a two node job to test resuming from an HF checkpoint? What's the process for the user? I imagine this will be a frequent request so I want to make sure it's as frictionless as possible.

I did a quick pass but will revisit later today. Thanks.

**Comment by hemildesai** [comment]
> @hemildesai this is great, have you run a two node job to test resuming from an HF checkpoint? What's the process for the user? I imagine this will be a frequent request so I want to make sure it's as frictionless as possible.
> 
> I did a quick pass but will revisit later today. Thanks.

Yes we will be updating the docs soon to have this information there.

**Review by akoumpa — DISMISSED** [review]
LGTM, thanks.

**Review by cuichenx — DISMISSED** [review]

**Review by cuichenx — APPROVED** [review]

**Review by cuichenx — APPROVED** [review]

**Comment by mlwu-ai** [comment]
> > @hemildesai this is great, have you run a two node job to test resuming from an HF checkpoint? What's the process for the user? I imagine this will be a frequent request so I want to make sure it's as frictionless as possible.
> > I did a quick pass but will revisit later today. Thanks.
> 
> Yes we will be updating the docs soon to have this information there.

when will the docs release in roadmap. The "import" command-line seems to be introduce in feature, however, the converted model could not be tensor partition or other method because the training strategy config is not recognized in `io.import_ckpt` function, how should I fix the problem?

**Pull request merged**